workspace/update-injected-packages/src/findInjectedPackages.ts

```diff
@@ -0,0 +1,49 @@
+import path from 'path'
+import * as dp from '@pnpm/dependency-path'
+import { type DirectoryResolution, type LockfileObject } from '@pnpm/lockfile.fs'
+import { type DepPath } from '@pnpm/types'
+
+export interface FindInjectedPackagesOptions {
+  lockfile: Pick<LockfileObject, 'packages'>
+  pkgName: string
+  pkgRootDir: string
+  virtualStoreDir: string
+  virtualStoreDirMaxLength: number
+  workspaceDir: string
+}
+
+export interface InjectedPackageInfo {
+  depPath: DepPath
+  parsedDepPath: dp.DependencyPath
+  resolution: DirectoryResolution
+  sourceDir: string
+  targetDir: string
+}
+
+export function * findInjectedPackages ({
```

Comment:
There is not need to get this info from the lockfile.

This info is present in `node_modules/.modules.yaml`

https://github.com/pnpm/pnpm/blob/49cc1672ff70c0d859dcc785766a02ba97a87523/pkg-manager/modules-yaml/src/index.ts#L34

Reply:
https://github.com/pnpm/pnpm/pull/9100/commits/16a527c045e6a06ec72939e71dc2d3128a5a5888